readthedocs/doc_builder/config.py

```diff
@@ -172,10 +164,27 @@ def load_yaml_config(version):
         # This is a subclass of ConfigError, so has to come first
         raise
     except ConfigError:
+        # TODO: this shouldn't be hardcoded here
+        env_config.update({
+            'output_base': '',
+            'type': 'sphinx',
+            'name': version.slug,
+        })
         config = BuildConfig(
             env_config=env_config,
             raw_config={},
             source_file='empty',
             source_position=0,
         )
+        config.validate()
```

Comment:
We weren't calling to `validate()` here (we were cheating!), but we had no problem because `raw_config` was empty, and `ConfigWrapper` was serving all default values from `version`. But now we are passing the defaults in the `env_config` parameter, so we need to call to validate (the values need to be initialized now).